plugins/sub_plugins/grpc/base.py

```diff
@@ -0,0 +1,65 @@
+# (c) 2022 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from functools import wraps
+
+from ansible.plugins import AnsiblePlugin
+
+
+def ensure_connect(func):
+    @wraps(func)
+    def wrapped(self, *args, **kwargs):
+        if not self._connection._connected:
+            self._connection._connect()
+        return func(self, *args, **kwargs)
+
+    return wrapped
+
+
+class GrpcBase(AnsiblePlugin):
+    """
+    A base class for implementing gRPC abstraction layer
+    """
+
+    __rpc__ = ["channel", "get_config", "edit_config", "get"]
+
+    def __init__(self, connection):
+        super(GrpcBase, self).__init__()
+        self._connection = connection
+
+    @property
```

Comment:
```suggestion
    @property
    @ensure_connect
```

iosxr is the only implementation linked, and it doesn't appear to use this property, but it looks like you need to be connected for this, and if we always go through `self.channel` we shouldn't need ensure_connect in the implementation